### PR TITLE
:card_file_box: Skill link to topic

### LINKED ISSUE #300: Possibility to link a skill to a topic
Please see seeds/07-skillTopics.sql

The admin should be link a skill to topic easily 
```
INSERT INTO public."SkillTopic" SELECT skill.id, topic.id FROM public."Topic" topic JOIN public."Skill" skill ON skill.name='Mob Programming' WHERE topic.name IN ('Craftsmanship', 'Development process') ON CONFLICT DO NOTHING;
```

### PR DESCRIPTION
**Pull request by MailyLehoux**
resolve #300

### CODE REVIEW

File: i18n/en.js [hunk 1]
```diff
@@ -26,6 +26,7 @@ export default {
     update: "Update",
     skillList: "Skills list",
     category: "Category",
+    topics: "Topics associated",
```
Comment (PierreBrethes) [inline i18n/en.js:29]: en anglais vaut mieux mettre l'adjectif en premier en général : Associated topics

File: src/components/SkillTopics.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      );
+    }
+  };
+
+  return (
+    <div
+      className={`flex flex-col rounded-lg ${
+        darkMode
+          ? "bg-light-dark dark:bg-dark-dark my-2 p-2"
+          : "bg-light dark:bg-dark-dark my-2 p-2"
```
Comment (PierreBrethes) [inline src/components/SkillTopics.tsx:62]: you should put my-2, p-2 and dark:bg-dark-dark outside of the darkmode condition as you will always have it
also dark:bg-dark-dark in darkmode and lightmode ? seems weird or good behavior ?

File: src/components/admin/EditSkillAdminModal.tsx [hunk 3]
```diff
@@ -34,11 +39,23 @@ const EditSkillAdminModal = ({
...
+      skillId: skill?.id,
+    },
+  });
 
   /*
    * MUTATIONS
    */
   const [editSkill] = useMutation<EditSkillMutation>(EDIT_SKILL);
+  const [addSkillToTopic] =
+    useMutation<AddSkillToTopicMutation>(ADD_SKILL_TO_TOPIC);
```
Comment (benjaminbrgs) [inline src/components/admin/EditSkillAdminModal.tsx:58]: This mutation is unused

File: src/components/SkillTopics.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    useQuery<GetTopicsInfosQuery>(GET_TOPICS_INFOS);
+
+  const [insertTopic] = useMutation(ADD_SKILL_TO_TOPIC);
+  const [deleteTopic] = useMutation(DELETE_SKILL_TO_TOPIC);
+
+  const updateTopic = (selectedTopic: { id: string; name: string }) => {
+    const skilltopic = skillTopics.SkillTopic?.find(
+      (value) => value.topicId === selectedTopic?.id
+    );
+    if (!skilltopic) {
```
Comment (benjaminbrgs) [inline src/components/SkillTopics.tsx:38]: For me, this is useless, as we already have a valid id. IDs received are predefined by the previous select query and then no user free input

**Review by benjaminbrgs — CHANGES_REQUESTED** [review]
Nice 👍🏻
It could be great to merge `PreferedTopic` and `SkillTopic` into one but it could be done in another issue

File: src/components/SkillTopics.tsx (hunk 4 above)
Comment (MailyLehoux) [reply src/components/SkillTopics.tsx:38]: I use this .find() two times : one for change css and another one to select the right request -> delete or add . For me this find is not useless but the process can be improve

**Comment by MailyLehoux** [comment]
Thanks guys for your reviews. 

For the bg dark and light I've opnened a new [issue](https://github.com/Zenika/skillZ/issues/314) . On the last commit, I've deleted the background. We didn't decided yet which color we have to use for this kind of component in light mode. It's no sense to have a background in dark mode but not in light

And so, I've opened this [issue](https://github.com/Zenika/skillZ/issues/315) to merge PreferedTopics and SkillTopics component. tell me if you have suggestions or whatever

File: src/components/SkillTopics.tsx (hunk 4 above)
Comment (benjaminbrgs) [reply src/components/SkillTopics.tsx:42]: Ok I got it. Maybe you can use the .find() only one time in the map() function (line 69), then use the result for css and to pass an updateTopic or deleteTopic method in the onClick prop ?

Comment (MailyLehoux) [reply src/components/SkillTopics.tsx:42]: hmm if we use this in the map, we would select only the prefered topics but we want all the topics for the render. I'm sorry I didn't find out an easy way to resolve it 💩 if you have the solution we can implement it, or we can add a commentary in the [issue](https://github.com/Zenika/skillZ/issues/315) that will refacto this component

Comment (benjaminbrgs) [reply src/components/SkillTopics.tsx:42]: I'll see what I can do in https://github.com/Zenika/skillZ/issues/315 otherwise I'll leave it like that 👍🏻

**Review by benjaminbrgs — APPROVED** [review]

**Review by PierreBrethes — APPROVED** [review]

**Pull request merged**